### DISCUSSION TITLE
Integrating Omega Template into existing Maven project: Menu not rendering, Menubutton not working.

### DISCUSSION THREAD

**Discussion by jakob-osl** [primefaces/community#1428] [PrimeFaces Templates]
Context:
I bought the Omega template for Primefaces with the goal to use it in my existing Spring/JSF/Primefaces application.
First of all, I didn't find good documentation for that and as I am not very familiar with Primefaces I was a bit overwhelmed.
However I got to get something working by following steps from an old dropbox tutorial:

This is what I did:
- download the omega zip archive
- added the omega-theme-6.0.0-jakarta.jar to my project (using mvn install)
- added the the resources folder (containing the folders omega-layout and sass) into my webapp folder
- added all the template files (template, sidebar...) to my WEB-INF folder
- added the remaining xhtml pages supplied by omega

The first problem I encountered was that template.xhtml was using the java class GuestPreferences, so I just added that as well to fix that.

Current problem:
I can start my application without any problems and view the page empty.xhtml.
However, on the left side the menu isnt showing and in the profile section, I dont see any profile links like "Account" or "Notifications" specified in the template for sidebar.
Additionally the button to toggle the sidebar isnt working, nothing happens when I press it.

![image](https://github.com/primefaces/community/assets/153545876/dbd9211b-be5b-4bd6-ac82-ba40bc50c13f)

Questions:
1. First of all, is my way of integrating Omega into my project the right way to do it? Is there any Documentation I wasn't able to find?

2. How can I debug my Problems? Where can I look for causes? I just don't know how I can solve this by my self as I also wasn't able to find anything helpful online any I am not very fluent in web development.

I would be very helpful if someone is able to help me because I find it a bit frustrating to buy such a product and then not being able to use it correctly. Maybe it's just my stupidity but necessary explanations / docs are missing in my opinion.

Thanks!

**Comment by NicolaIsotta** [comment]
Have you followed the Omega documentation? https://primefaces.org/omega/documentation.xhtml
If you have followed the docs and there's still something broken, you could look in the server logs or in the browser console.

**Reply by jakob-osl** [reply]
I read the documentation but I didn't follow it. This documentation only describes how to run the sample demo, not how to use it generally with an arbitrary (already existing) project or at least explain to me in a reasonable way how the Omega them/layout is working in general. This was not very helpful for me. I neither have errors in the server logs nor the browser console. It might be that I am missing some file or component, but as I described it is not very clear to me how to integrate Omega into my project (or to debug the occuring problems)

**Reply by melloware** [reply]
@jakob-osl yep but you could start with the sample demo code all working and then strip out anything you don't need.

**Reply by jakob-osl** [reply]
I don't want to do that as this is not suitable for me. I already have a project setup which I am not supposed to change and I am not 
100%  familiar with that to change all the configs etc. I thought if I buy that Omega theme I would be able to integrate all the necessary parts. Even though there is no good explanations / complete documentation to be found. I still managed to get the main parts to run (as explained in my original post) but I still have some problems (as already described. If nobody can answer my questions or provide me with general documentation I think I just have to scrap that idea of using omega. Running the sample demo and doing pretty much all the work to get it somehow running was not what I was expecting when I bought a theme / layout. Is it relatable what I am saying or am I just stupid?

**Reply by melloware** [reply]
No its relatable what you are saying.  I think to do what you want to do is perfectly fine its just you are expecting a drop in solution but a few things to note.

1. You mentioned Spring, are you using Spring Boot or JoinFaces or just a vanilla Spring MVC app in Tomcat?  Because JSF can be quirky with some of those frameworks you may have to make slight changes to the theme like where resources are stored. I was using Babylon with JoinFaces and I remember having to tweak the theme to work the "JoinFaces way".

2. In Chrome F12 console are you seeing any errors?  The fact the menu is not opening and closing the sidebar tells me some JS code is missing?

**Comment by tandraschko** [comment]
i will add docs for integrating it into a existing maven application for the next release
